internal/metrics/client_test.go

```diff
@@ -0,0 +1,665 @@
+				t.Errorf("expected %d responses, got %d", len(tt.expectedResponses), len(responses))
+				return
+			}
+
+			for i, expected := range tt.expectedResponses {
+				if responses[i].Key != expected.Key {
+					t.Errorf("response[%d]: expected key %q, got %q", i, expected.Key, responses[i].Key)
+				}
+				if math.IsNaN(expected.Value) {
+					if !math.IsNaN(responses[i].Value) {
+						t.Errorf("response[%d]: expected NaN value, got %f", i, responses[i].Value)
+					}
+				} else if responses[i].Value != expected.Value {
+					t.Errorf("response[%d]: expected value %f, got %f", i, expected.Value, responses[i].Value)
+				}
+			}
+		})
+	}
+}
+
+func TestLocalMetricsClient_GetMetrics(t *testing.T) {
+	tests := []struct {
+		name              string
+		metrics           []parser.Metric
+		metricsData       string
+		expectedResponses []MetricsResponse
+		expectedError     string
+	}{
+		{
+			name:              "empty metric list",
+			metrics:           []parser.Metric{},
+			metricsData:       "",
+			expectedResponses: []MetricsResponse{},
+		},
+		{
+			name: "exact metric match",
+			metrics: []parser.Metric{
+				{Name: "test_metric", Query: ""},
+			},
+			metricsData: "test_metric{label=\"value\"} 42.5",
+			expectedResponses: []MetricsResponse{
+				{Key: "test_metric", Value: 42.5},
+			},
+		},
+		{
+			name: "no partial match",
+			metrics: []parser.Metric{
+				{Name: "test", Query: ""},
+			},
+			metricsData:       "test_metric{label=\"value\"} 42.5\ntest_another{label=\"value\"} 10.0",
+			expectedResponses: []MetricsResponse{},
+		},
+		{
+			name: "multiple exact matches",
+			metrics: []parser.Metric{
+				{Name: "metric1", Query: ""},
+				{Name: "metric2", Query: ""},
+			},
+			metricsData: "metric1{label1=\"value1\",label2=\"value2\"} 42.5\nmetric2{label=\"value\"} 10.0",
+			expectedResponses: []MetricsResponse{
+				{Key: "metric1", Value: 42.5},
+				{Key: "metric2", Value: 10.0},
+			},
+		},
+		{
+			name: "invalid metric format",
+			metrics: []parser.Metric{
+				{Name: "test", Query: ""},
+			},
+			metricsData:   "test_metric invalid_value",
+			expectedError: "failed to parse metric value",
+		},
+		{
+			name: "missing metric value",
+			metrics: []parser.Metric{
+				{Name: "test", Query: ""},
+			},
+			metricsData:       "test_metric{label=\"value\"}",
+			expectedResponses: []MetricsResponse{},
+		},
+		{
+			name: "non-numeric metric value",
+			metrics: []parser.Metric{
+				{Name: "test_metric", Query: ""},
+			},
+			metricsData: "test_metric{label=\"value\"} NaN",
+			expectedResponses: []MetricsResponse{
+				{Key: "test_metric", Value: math.NaN()},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a test server that returns the metrics data
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "text/plain")
+				w.Write([]byte(tt.metricsData))
```

Comment:
It could be convenient to create an actual Prometheus server if you want to make this a bit more robust. Not required of course, just a thought.

Reply:
Added a TODO for the beautification phase after the adoption phase :)